shadowsocks/udp.go

```diff
@@ -175,48 +175,54 @@ func (s *udpService) Stop() error {
 	return s.clientConn.Close()
 }
 
+type natentry struct {
+	conn           net.PacketConn
+	clientLocation string
```

Comment:
Please add a comment saying that we cache the client location to avoid a geo ip lookup for every packet.

Reply:
Done